src/Microsoft.AspNet.Mvc.Core/ModelBinding/KeyValuePairModelBinder.cs

```diff
@@ -62,8 +62,27 @@ public sealed class KeyValuePairModelBinder<TKey, TValue> : IModelBinder
             }
             else
             {
-                // Caller (GenericModelBinder) was able to resolve a binder type and will create a ModelBindingResult
-                // that exits current ModelBinding loop.
+                // If this is the fallback case, and we failed to find data as a top-level model, then generate a
+                // default 'empty' model and return it.
```

Comment:
Odd case. Curious whether Web API 2 or MVC 5 pass `default(KeyValuePair<TKey, TValue>)` into an action. Of course it's fine to support here.